### PR TITLE
Refactors signallers, assemblies, electropacks

### PR DESCRIPTION
**Pull request by AffectedArc07**
### AA Note

The amount of times I had to type `signaler` instead of `signaller` almost killed me.

## What Does This PR Do

Part of #18685 

Refactors signallers to use a global list instead of the radio signal awful system. This also meant refactoring electropacks, lightly refactoring the electric chair, as well as PDAs and PAIs to account for the new system. It also converts signallers to TGUI, that way I can re-use the UI for them.

I also ended up having to lightly refactor assemblies to make the examine text work, but I may as well fix it.

## Why It's Good For The Game
I will kill this radio system. Its so badly designed and has so much overhead it may as well not exist.

Below is a list of things that should use this radio system:
- Radios

This whole thing needs cleaning up, and I will do that. The performance gains and simplifications will be nice.

## Testing
- Tested signallers on their own
- Tested signallers in grenades in assemblies
- Tested signallers on a TTV
- Tested the electric chair
- Tested electropacks

## Changelog
:cl:
add: Remote signallers now use TGUI
/:cl:

### CODE REVIEW

File: code/modules/assembly/signaler.dm [hunk 1]
```diff
@@ -6,147 +8,92 @@
...
+	. = ..()
+	. += "The power light is [receiving ? "on" : "off"]"
+
+// Called from activate(), actually invokes the signal on other signallers in the world
+/obj/item/assembly/signaler/proc/signal()
+	for(var/obj/item/assembly/signaler/S as anything in GLOB.remote_signalers)
+		if(S == src)
+			continue
+		if(S.receiving && (S.code == code) && (S.frequency == frequency))
+			S.signal_callback()
```
Comment (AffectedArc07) [inline code/modules/assembly/signaler.dm:38]: To answer the "Why dont we check this on each signaler itself inside the `signal_callback()` proc?" question.

1. Overriding `signal_callback()` to just do the action as opposed to sanity checks is a lot easier to keep track of
2. By doing the checks here, it saves the time that would be jumping between procs and this loop and back, which can consume more CPU than you would think. 

Yes I am microoptimising this much.

File: code/game/objects/items/devices/radio/electropack.dm [hunk 2]
```diff
@@ -1,43 +1,57 @@
...
+	integrated_signaler = new /obj/item/assembly/signaler/electropack(src, src) // Loc and the integrated one
 
-/obj/item/radio/electropack/Destroy()
-	if(istype(src.loc, /obj/item/assembly/shock_kit))
+/obj/item/electropack/Destroy()
+	integrated_signaler.owning_pack = null
+	QDEL_NULL(integrated_signaler)
+
+	if(istype(loc, /obj/item/assembly/shock_kit))
 		var/obj/item/assembly/shock_kit/S = src.loc
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/radio/electropack.dm:23]:
```suggestion
		var/obj/item/assembly/shock_kit/S = loc
```

File: code/game/objects/items/devices/radio/electropack.dm [hunk 3]
```diff
@@ -1,43 +1,57 @@
...
+	if(src == user.back)
+		to_chat(user, "<span class='notice'>You need help taking this off!</span>")
+		return FALSE
+
+	..()
+
+/obj/item/electropack/attack_self(mob/user)
+	ui_interact(user)
+
+/obj/item/electropack/attackby(obj/item/W as obj, mob/user as mob, params)
```
Comment (hal9000PR) [inline code/game/objects/items/devices/radio/electropack.dm:44]:
```suggestion
/obj/item/electropack/attackby(obj/item/W, mob/user, params)
```

File: code/modules/assembly/signaler.dm [hunk 4]
```diff
@@ -6,147 +8,92 @@
...
+	GLOB.remote_signalers -= src
 	return ..()
 
-/obj/item/assembly/signaler/describe()
-	return "[src]'s power light is [receiving ? "on" : "off"]"
+/obj/item/assembly/signaler/examine(mob/user)
+	. = ..()
+	. += "The power light is [receiving ? "on" : "off"]"
+
+// Called from activate(), actually invokes the signal on other signallers in the world
```
Comment (hal9000PR) [inline code/modules/assembly/signaler.dm:32]:
```suggestion
/// Called from activate(), actually invokes the signal on other signallers in the world
```

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**